addons/account/models/account_move_line.py

```diff
@@ -1580,10 +1580,10 @@ def write(self, vals):
         violated_fields = set(vals) & inalterable_fields
         if hashed_moves and violated_fields:
             raise UserError(_(
-                "You cannot edit the following fields: %s.\n"
-                "The following entries are already hashed:\n%s",
-                ', '.join(f['string'] for f in self.fields_get(violated_fields).values()),
-                '\n'.join(hashed_moves.mapped('name')),
+                "You cannot edit the following fields: %(fields)s.\n"
+                "The following entries are already hashed:\n%(entries)s",
+                fields=format_list(self.env, [f['string'] for f in self.fields_get(violated_fields).values()]),
```

Comment:
Just a suggestion, can't we use the same mechanism as for `_()` to retrieve the lang instead of providing the environment explicitly?

Reply:
Not a bad suggestion, and it would make sense for format_list and _ to use the same mechanism. Noted 🙂